internal/operands/common-templates/reconcile.go

```diff
@@ -108,7 +117,13 @@ func (c *commonTemplates) Reconcile(request *common.Request) ([]common.ResourceS
 	funcs = append(funcs, oldTemplateFuncs...)
 	funcs = append(funcs, reconcileTemplatesFuncs(request)...)
 
-	return common.CollectResourceStatus(request, funcs...)
+	results, err := common.CollectResourceStatus(request, funcs...)
+	for _, r := range results {
```

Comment:
The `results` contains results not only from created templates, but also from old templates (leftovers from old ssp operator version), and non-template resources like roles and role bindings. See the definition of `funcs` on line 77.

You could split the `funcs` list and call `CollectResourceStatus()` on parts of it, and then increase the metric only for template resources:

```go
funcs = append(funcs, oldTemplateFuncs...)
results, err := common.CollectResourceStatus(request, funcs...)
if err != nil {
	return nil, err
}

resultsTemplates, err := common.CollectResourceStatus(request, reconcileTemplatesFuncs(request)...)
if err != nil {
	return nil, err
}
for _, r := range resultsTemplates {
	if r.OperationResult == controllerutil.OperationResultUpdated {
		CommonTemplatesRestored.Inc()
	}
}
return append(results, resultsTemplates...), nil
```

Reply:
yep!